README.md

```diff
@@ -153,113 +154,119 @@ We know that we want to yield each element of the array successively to a block
+The first time through our `while` loop, `i` is equal to `0`. The second time through the loop, `i` is `1`, and so on. This will continue until `i` is equal to the index number of the last item in our array.
 
-So, inside each step of the `while` loop, `i` equals a given index number of our array. We can use this information to yield each successive value stored in our array, to our block. 
+So, during each step of the `while` loop, `i` equals a given index number of our array. We can use this information to yield each successive value stored in the array to the passed-in block:
 
 ```ruby
 def hello_t(array)
   i = 0
+
   while i < array.length
     yield array[i]
-		i = i + 1
+    i = i + 1
   end
 end
 ```
 
-Here, we use the `[]`, bracket, method to grab the value of each successive index element as we proceed through our `while` loop, and `yield` it to a block. Now we're ready to call our method:
+Here, we use the bracket (`[]`) method to grab the value of each successive index element as we proceed through our `while` loop, `yield`ing each to a block. Now we're ready to call our method:
 
-#### Step 4: Calling Our Method
+#### Step 4: Calling our method
 
-Let's call our method with an argument of the following array:
+Let's invoke our method, passing in the following array as an argument:
 
 ```ruby
 ["Tim", "Tom", "Jim"]
 ```
 
 ```ruby
 hello_t(["Tim", "Tom", "Jim"]) do |name|
-	if name.start_with?("T")
-		puts "Hi, #{name}"
-	end
+  if name.start_with?("T")
+    puts "Hi, #{name}"
+  end
 end
 ```
 
-Copy and paste the above method call into your `lib/hello.rb` file and run the file with `ruby lib/hello.rb` in your terminal. You should see the following:
+Copy and paste the above method call into your `lib/hello.rb` file (paste it below the line that reads `# call your method here!`). Then, run the file with the `ruby lib/hello.rb` command in your terminal. You should see the following:
 
 ```bash
 Hi, Tim
 Hi, Tom
 ```
 
-We call our method with the argument of our array and accompany that method call with a block that contains the `name` parameter and the desired code to `puts` out a greeting if the person's name begins with the letter `"T"`. Good job!
+We're invoking our method with the array of names as an argument and accompanying that method call with a block that accepts a `|name|` parameter. If the passed-in name begins with the letter `"T"`, the block will `puts` out a greeting. Good job!
 
-#### Step 5: Passing Our Test
+#### Step 5: Passing our test
 
-Go ahead and run the test suite by typing `learn` into your terminal in the directory of this lesson. You'll see that our test is expecting us to return the original array but our method is currently returning nil.
+Go ahead and run the test suite by typing `learn` into your terminal in this lesson's directory. You'll see that our test expects us to return the original array, but our method is currently returning `nil`.
```

Comment:
Show the error message that makes you believe that fact. Getting students to understand error messages is a constant battle. Any time we can show instead of tell, we should